### PR TITLE
Refactor SSD: Move everything button related to `src/ssd/ssd button.c`

### PR DESCRIPTION
**Pull request by Consolatis**
Follow-up from
- #1045 

No functional changes intended ~~, will rebase once #1045 is merged.~~

Edit:
This PR also adds a new node descriptor to the rounded button root tree node which has pointers for the rounded corner buffer and a `struct ssd_button`. That allows to decouple the actual buttons (e.g. `menu` or `close`) from the rounded corners which in turn allows for a future implementation of `titleLayout`.

### CODE REVIEW

File: src/ssd/ssd_button.c [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+#include "common/mem.h"
+#include "common/scene-helpers.h"
+#include "config/rcxml.h"
+#include "labwc.h" /* for MIN() */
+#include "node.h"
+#include "ssd-internal.h"
+
+struct ssd_rounded {
+	struct ssd_button *button;
+	struct wlr_scene_node *node;
```
Comment (Consolatis) [inline src/ssd/ssd_button.c:15]: `s/node/rounded_buffer_node/`

**Comment by ludg1e** [comment]
Hello, may I ask, what is left here for this to be ready to be merged?
May I guess, the separation in parts of the file ssd_part.c?

I'd be willing to work (and try to finish) this PR if I'd know what remains left to be done

Thank you and have a nice day

**Comment by Consolatis** [comment]
The main issue here is that we changed quite a lot in this area and this now needs to be rebased. It might be easier to start fresh using this is PR a mere template (e.g. which functions to move). There are further changes intended so we may want to wait a bit with the refactor to reduce further conflicts.

**Comment by ludg1e** [comment]
Related: #1334, #1296 
Other: #1018

**Comment by Consolatis** [comment]
If somebody wants to take this over feel free.

**Pull request closed without merging**